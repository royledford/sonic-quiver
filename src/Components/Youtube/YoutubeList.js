import React, { Component } from 'react'
import PropTypes from 'prop-types'
import YoutubeService from '../../api/youtubeApi'
import YoutubeSmall from './YoutubeSmall'
import YoutubePlayer from './YoutubePlayer'
import './YoutubeList.css'

export default class YoutubeList extends Component {
  static propTypes = {
    onClick: PropTypes.func,
  }

  constructor(props) {
    super(props)
    this.state = {
      ytVids: [],
      loading: false,
      videoToPlay: '',
      videoTitle: '',
      showVideo: false,
    }
    this.handlePlayClicked = this.handlePlayClicked.bind(this)
  }

  componentWillMount() {
    this.setState({ loading: true })
    YoutubeService.getVideos().then(vids => {
      this.setState({ ytVids: vids.items, loading: false })
    })
  }

  handlePlayClicked(id, title) {
    this.setState({
      videoToPlay: id,
      videoTitle: title,
      showVideo: true,
    })
  }

  handlePlayerClosed = () => {
    this.setState({
      videoToPlay: '',
      showVideo: false,
    })
  }

  render() {
    const { videoToPlay, videoTitle, showVideo, ytVids } = this.state
    return (
      <div className="youtubelist-wrap">
        <div className="youtubelist-row">
          <div className="youtubelist-list">
            {ytVids.map(
              vid =>
                vid.id.videoId ? (
                  <YoutubeSmall
                    key={vid.id.videoId}
                    id={vid.id.videoId}
                    title={vid.snippet.title}
                    description={vid.snippet.description}
                    thumbnailUrl={vid.snippet.thumbnails.high.url}
                    playVideo={this.handlePlayClicked}
                  />
                ) : null
            )}
          </div>
        </div>
        <YoutubePlayer
          videoId={videoToPlay}
          title={videoTitle}
          show={showVideo}
          onBackgroundClick={this.handlePlayerClosed}
          autoplay={1}
        />
      </div>
    )
  }
}
