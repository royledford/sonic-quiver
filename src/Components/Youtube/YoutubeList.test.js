import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import YoutubeList from './YoutubeList'
import renderer from 'react-test-renderer'

it('renders', () => {
  const tree = renderer.create(<YoutubeList />).toJSON()
  expect(tree).toMatchSnapshot()
})
