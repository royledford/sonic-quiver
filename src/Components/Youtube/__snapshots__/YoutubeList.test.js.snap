// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="youtubelist-wrap"
>
  <div
    className="youtubelist-row"
  >
    <div
      className="youtubelist-list"
    />
  </div>
</div>
`;
